src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs

```diff
@@ -145,14 +142,6 @@ private bool ShouldStartNewInstance(ImmutableHashSet<string> requiredPackageIds)
                 || !_currentlyRunningInstance.IsRunning;
         }
 
-        private void ThrowExceptionIfAlreadyHasActiveContext()
-        {
-            if (_hasCurrentlyActiveContext)
-            {
-                throw new Exception($"The previous integration test failed to call {nameof(VisualStudioInstanceContext)}.{nameof(Dispose)}. Ensure that test does that to ensure the Visual Studio instance is correctly cleaned up.");
```

Comment:
Is it not beneficial to fix `_hasCurrentlyActiveContext` instead? The original goal was to ensure any test specific resources/etc were properly cleaned up. This was meant to help ensure that one test failure caused a new instance of VS to come up (rather than letting the next test start in an unknown state).

Reply:
I agree with you in concept. However, it's not uncommon to encounter situations where a system-related failure will cascade to all tests failing. Each restart takes 60-90 seconds, so with 400 tests you're guaranteed to hit the global test timeout. The downside of the global test timeout is it doesn't capture the xunit log for the tests run prior to the timeout, so you're working with limited information.

Note that this only applies to tests which fail in such a way that the IDE state reset can't figure out how to get back to a consistent start state. This is not generally the case, so for most types of test failure it will continue with the next test as you expect.